src/psyclone/domain/common/transformations/alg_invoke_2_psy_call_trans.py

```diff
@@ -147,47 +147,39 @@ def remove_imported_symbols(node):
             :py:class:`psyclone.domain.common.algorithm.AlgorithmInvokeCall`
 
         '''
-        # Get a unique set of kernel functor symbols for this invoke
-        # if they are explicitly imported.
-        kernel_functor_symbols = set()
-        for kernel_functor in node.children:
-            if kernel_functor.symbol.is_import:
-                kernel_functor_symbols.add(kernel_functor.symbol)
-        # Remove imported symbols as appropriate
-        for kernel_functor_symbol in kernel_functor_symbols:
-            # Is this kernel_functor used in a different invoke?
-            used_elsewhere = False
-            # Search from where the symbol is declared
-            kf_symbol_table = kernel_functor_symbol.find_symbol_table(node)
-            scope_node = kf_symbol_table.node
-            for invoke in scope_node.walk(AlgorithmInvokeCall):
-                if invoke != node:
-                    for kernel_functor in invoke.children:
-                        if kernel_functor.symbol == kernel_functor_symbol:
-                            used_elsewhere = True
-                            break
-                if used_elsewhere:
-                    break
-
-            if not used_elsewhere:
-                # remove the symbol (and, potentially, the container
-                # from which it is imported) from the symbol table.
-                container_symbol = \
-                    kernel_functor_symbol.interface.container_symbol
-                c_symbol_table = container_symbol.find_symbol_table(node)
-                # issue #898 not currently possible to remove a
-                # DataTypeSymbol using the remove method.
-                # pylint: disable=protected-access
-                norm_name = c_symbol_table._normalize(
-                    kernel_functor_symbol.name)
-                c_symbol_table._symbols.pop(norm_name)
-                # pylint: enable=protected-access
-                try:
-                    c_symbol_table.remove(container_symbol)
-                except ValueError:
-                    # container symbol still imports one or more symbols
-                    # so can not be removed.
-                    pass
+        all_functors = node.root.walk(KernelFunctor)
+        local_functor_symbols = set()
+        remote_functor_symbols = set()
+        for functor in all_functors:
+            if not functor.symbol.is_import:
+                continue
```

Comment:
Please add a comment: "If a KernelFunctor is not imported then it must be a BuiltIn"

Reply:
Added.